base/thread_utils.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <thread>
+#include <vector>
+
+#include "base/macros.hpp"
+
+namespace base
+{
+namespace threads
+{
+template <typename Thread = std::thread, typename ThreadContainer = std::vector<Thread>>
+class ThreadsJoiner
+{
+public:
+  explicit ThreadsJoiner(ThreadContainer & threads) : m_threads(threads) {}
+
+  ~ThreadsJoiner()
+  {
+    for (auto & thread : m_threads)
+    {
+      if (thread.joinable())
+        thread.join();
+    }
+  }
+
+private:
+  ThreadContainer & m_threads;
+};
+
+// This class is needed in ThreadPool to store std::packaged_task<> objects.
```

Comment:
computational:: ?

Reply:
or just "in thread pools" and s/is needed/can be used/

Reply:
++